scripts/src/java/org/oppia/android/scripts/coverage/RunCoverage.kt

```diff
@@ -0,0 +1,154 @@
+import org.oppia.android.scripts.proto.TestFileExemptions
+import java.io.File
+import java.io.FileInputStream
+import java.util.concurrent.TimeUnit
+
+/**
+ * Entry point function for running coverage analysis for a source file.
+ *
+ * Usage:
+ *   bazel run //scripts:run_coverage_for_test_target -- <path_to_root> <relative_path_to_file>
+ *
+ * Arguments:
+ * - path_to_root: directory path to the root of the Oppia Android repository.
+ * - relative_path_to_file: the relative path to the file to analyse coverage
+ *
+ * Example:
+ *     bazel run //scripts:run_coverage -- $(pwd)
+ *     utility/src/main/java/org/oppia/android/util/parser/math/MathModel.kt
+ * Example with custom process timeout:
+ *     bazel run //scripts:run_coverage -- $(pwd)
+ *     utility/src/main/java/org/oppia/android/util/parser/math/MathModel.kt processTimeout=10
+ *
+ */
+fun main(vararg args: String) {
+  val repoRoot = args[0]
+  val filePath = args[1]
+
+  ScriptBackgroundCoroutineDispatcher().use { scriptBgDispatcher ->
+    val processTimeout: Long = args.find { it.startsWith("processTimeout=") }
+      ?.substringAfter("=")
+      ?.toLongOrNull() ?: 5
+
+    val commandExecutor: CommandExecutor = CommandExecutorImpl(
+      scriptBgDispatcher, processTimeout = processTimeout, processTimeoutUnit = TimeUnit.MINUTES
+    )
+
+    RunCoverage(repoRoot, filePath, commandExecutor, scriptBgDispatcher).execute()
+  }
+}
+
+/**
+ * Class responsible for executing coverage on a given file.
+ *
+ * @param repoRoot the root directory of the repository
+ * @param filePath the relative path to the file to analyse coverage
+ * @param commandExecutor Executes the specified command in the specified working directory
+ * @param scriptBgDispatcher the [ScriptBackgroundCoroutineDispatcher] to be used for running the coverage command
+ */
+class RunCoverage(
+  private val repoRoot: String,
+  private val filePath: String,
+  private val commandExecutor: CommandExecutor,
+  private val scriptBgDispatcher: ScriptBackgroundCoroutineDispatcher
+) {
+  private val bazelClient by lazy { BazelClient(File(repoRoot), commandExecutor) }
+
+  private val rootDirectory = File(repoRoot).absoluteFile
+  private val testFileExemptionTextProto = "scripts/assets/test_file_exemptions"
+
+  /**
+   * Executes coverage analysis for the specified file.
+   *
+   * Loads test file exemptions and checks if the specified file is exempted. If exempted,
+   * prints a message indicating no coverage analysis is performed. Otherwise, initializes
+   * a Bazel client, finds potential test file paths, retrieves Bazel targets, and initiates
+   * coverage analysis for each test target found.
+   */
+  fun execute(): MutableList<List<String>> {
+    var coverageDataList = mutableListOf<List<String>>()
+    val testFileExemptionList = loadTestFileExemptionsProto(testFileExemptionTextProto)
+      .getExemptedFilePathList()
+
+    val isExempted = testFileExemptionList.contains(filePath)
+    if (isExempted) {
```

Comment:
```suggestion
    if (filePath in testFileExemptionList) {
```

Can be simplified a bit.

Reply:
Done! Simplified as mentioned.